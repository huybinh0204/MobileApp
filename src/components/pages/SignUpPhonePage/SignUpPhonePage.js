import { yupResolver } from '@hookform/resolvers/yup';
import { useNavigation, useRoute } from '@react-navigation/native';
import React, { useEffect, useState } from 'react';
import { useForm } from 'react-hook-form';
import { Keyboard, LayoutAnimation } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { number, object } from 'yup';
import { Form, FormInput, MainButton, Toast } from '_components/atoms';
import { OnboardingScreenLayout } from '_components/organisms';
import { NAVIGATION, PHONE_CODES, REGEX, TOAST_TYPES } from '_constants';
import strings from '_localization/index';
import { CustomerActions, CustomerSelectors } from '_store/customer';
import { cleanPhoneNumber, formatPhoneNumber } from '_utilities/Account';
import {
  Container,
  Disclosure,
  DisclosureContainer,
  ErrorMessage,
  Hyperlink,
  InputRow,
  PhoneNumberTitle,
  Row,
} from './SignUpPhonePage.styles';

const schema = object().shape({
  phoneNumber: number().min(10).max(10),
});

const defaultValues = {
  phoneNumber: '',
};

const SignUpPhonePage = () => {
  const { params } = useRoute();
  const { navigate } = useNavigation();
  const dispatch = useDispatch();

  const customerExternalId = useSelector(CustomerSelectors.getExternalId);
  const codeSuccess = useSelector(CustomerSelectors.getPhoneVerificationSuccess);
  const codeError = useSelector(CustomerSelectors.getPhoneVerificationError);
  const isLoading = useSelector(CustomerSelectors.getIsLoading);

  const [error, setError] = useState(null);
  const [isDisclosureVisible, setIsDisclosureVisible] = useState(true);

  const formMethods = useForm({ defaultValues, resolver: yupResolver(schema) });

  const { watch } = formMethods;
  const phoneNumberValue = watch('phoneNumber');

  useEffect(() => {
    if (codeSuccess) {
      dispatch(CustomerActions.setPhoneVerificationError(null));
      dispatch(CustomerActions.setPhoneVerificationSuccess(false));
      navigate(NAVIGATION.auth.signUpPhoneConfirmation, {
        ...params,
        phoneNumber: phoneNumberValue,
      });
    }
  }, [codeSuccess, navigate, params, phoneNumberValue, dispatch]);

  const setDisclosureVisibility = (value) => {
    LayoutAnimation.easeInEaseOut();
    setIsDisclosureVisible(value);
  };

  const handleSubmit = () => {
    if (phoneNumberValue.match(REGEX.phoneNumber)) {
      setError(null);
      dispatch(
        CustomerActions.startPhoneVerification(
          customerExternalId,
          cleanPhoneNumber(PHONE_CODES.US, phoneNumberValue)
        )
      );
      Keyboard.dismiss();
    } else {
      setError(strings.signUp.phoneNumber.invalid);
    }
  };

  const handleCloseToast = () => {
    dispatch(CustomerActions.setPhoneVerificationError(null));
    dispatch(CustomerActions.setPhoneVerificationSuccess(false));
  };

  return (
    <OnboardingScreenLayout step={4} testID="SignUpPhonePage" showBackButton={false}>
      <Container>
        <PhoneNumberTitle>{strings.signUp.phoneNumber.title}</PhoneNumberTitle>
        <Form formMethods={formMethods}>
          <Row>
            <InputRow flex={1} marginRight>
              <FormInput
                autoCapitalize="none"
                keyboardType="phone-pad"
                name="phone_code"
                value={PHONE_CODES.US}
                placeholder={'Code'}
                testID="phoneCode"
                editable={false}
              />
            </InputRow>
            <InputRow flex={4}>
              <FormInput
                autoCapitalize="none"
                keyboardType="phone-pad"
                name="phoneNumber"
                placeholder={strings.signUp.phoneNumber.placeholder}
                testID="phoneNumber"
                textContentType="telephoneNumber"
                maxLength={14}
                value={formatPhoneNumber(phoneNumberValue)}
                onFocus={() => setDisclosureVisibility(false)}
                onBlur={() => setDisclosureVisibility(true)}
              />
            </InputRow>
          </Row>
          <ErrorMessage>{error}</ErrorMessage>
          <DisclosureContainer>
            {isDisclosureVisible && (
              <Disclosure>
                {strings.formatString(strings.signUp.phoneNumber.disclosure, {
                  privacyPolicy: (
                    <Hyperlink href={strings.settings.policyTerms.kinlyUser}>
                      {strings.signUp.phoneNumber.privacyPolicy}
                    </Hyperlink>
                  ),
                })}
              </Disclosure>
            )}
          </DisclosureContainer>
          <MainButton
            disabled={phoneNumberValue === ''}
            isLoading={isLoading}
            onPress={handleSubmit}
            testID="signUpPhoneSubmitButton"
          >
            {strings.signUp.continue}
          </MainButton>
        </Form>
      </Container>
      <Toast
        type={TOAST_TYPES.ERROR}
        header={codeError?.title}
        content={codeError?.description}
        onClose={handleCloseToast}
        show={codeError !== null}
        testID="codeErrorToast"
      />
    </OnboardingScreenLayout>
  );
};

export default SignUpPhonePage;
