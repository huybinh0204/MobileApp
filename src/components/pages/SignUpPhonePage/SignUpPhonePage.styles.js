import styled from '@emotion/native';
import { BaseText } from '_components/atoms';

export const Container = styled.View(({ theme }) => ({
  flex: 1,
  padding: theme.spacing.m,
}));

export const PhoneNumberTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.m,
}));

export const ErrorMessage = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.epsilon500,
  marginTop: theme.spacing.xs,
  textAlign: 'left',
}));

export const DisclosureContainer = styled.View({
  marginTop: 'auto',
});

export const Disclosure = styled(BaseText)(({ theme }) => ({
  ...theme.typography.legal,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));

export const Hyperlink = styled(Disclosure)(({ theme }) => ({
  color: theme.colors.alpha500,
}));

export const PhoneInputContainer = styled.View(() => ({
  flexDirection: 'row',
  alignItems: 'center',
  width: '100%',
  justifyContent: 'space-between',
}));

export const Row = styled.View({
  flexDirection: 'row',
});

export const InputRow = styled.View(({ flex, theme, marginRight }) => ({
  flex,
  marginRight: marginRight ? theme.spacing.s : 0,
}));
