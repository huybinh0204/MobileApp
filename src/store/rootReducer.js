import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
