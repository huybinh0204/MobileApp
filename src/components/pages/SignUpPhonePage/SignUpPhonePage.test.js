import { useNavigation } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import * as ReactRedux from 'react-redux';
import { createStore } from 'redux';
import { NAVIGATION } from '_constants/index';
import strings from '_localization/index';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import SignUpPhonePage from './SignUpPhonePage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

const dispatch = jest.fn();
jest.spyOn(ReactRedux, 'useDispatch').mockReturnValue(dispatch);

describe('SignUpPhonePage', () => {
  let mockStore;
  let mockNavigation;

  beforeEach(() => {
    mockStore = createStore(rootReducer, {
      auth: {
        credentials: {
          sub: 'identity-unique-id',
          accessToken: 'eyJhbGciOiJSUzIYzZnpabjNoNnSJ9.eyJpc3M',
        },
      },
      customer: {
        externalId: 'mocked-external-id',
      },
    });
    mockNavigation = navigationMock(jest.fn);
    useNavigation.mockImplementation(() => mockNavigation);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should dispatch an action if the phone number is correct', async () => {
    const { getByText, getByTestId } = render(
      <ComponentWithProviders component={SignUpPhonePage} store={mockStore} />
    );

    const phoneNumberInput = getByTestId('phoneNumber');
    fireEvent.changeText(phoneNumberInput, '5104831826');

    const submitButton = getByText(strings.signUp.continue);

    await waitFor(() => {
      fireEvent.press(submitButton);
    });

    expect(dispatch).toBeCalledWith({
      customerExternalId: 'mocked-external-id',
      phoneNumber: '+15104831826',
      type: '@@BE-CUSTOMER/START_PHONE_VERIFICATION',
    });
  });

  it('should trigger the change of screen if the code was requested successfully', async () => {
    mockStore = createStore(rootReducer, {
      auth: {
        credentials: {
          sub: 'identity-unique-id',
          accessToken: 'eyJhbGciOiJSUzIYzZnpabjNoNnSJ9.eyJpc3M',
        },
      },
      customer: {
        phoneVerificationSuccess: true,
      },
    });

    const { getByText, getByTestId } = render(
      <ComponentWithProviders component={SignUpPhonePage} store={mockStore} />
    );

    const phoneNumberInput = getByTestId('phoneNumber');
    fireEvent.changeText(phoneNumberInput, '5104831826');

    const submitButton = getByText(strings.signUp.continue);

    await waitFor(() => {
      fireEvent.press(submitButton);
    });

    expect(mockNavigation.navigate).toBeCalledWith(NAVIGATION.auth.signUpPhoneConfirmation, {
      phoneNumber: '5104831826',
    });
  });

  it('should not navigate if some field is missing', async () => {
    const { getByText } = render(
      <ComponentWithProviders
        component={SignUpPhonePage}
        store={mockStore}
        params={{ firstName: 'First', lastName: 'Last', dateOfBirth: '09241997' }}
      />
    );

    const submitButton = getByText(strings.signUp.continue);

    await waitFor(() => {
      fireEvent.press(submitButton);
    });

    expect(mockNavigation.navigate).toBeCalledTimes(0);
  });

  it('should match snapshot', () => {
    const { getByTestId, toJSON } = render(
      <ComponentWithProviders
        component={SignUpPhonePage}
        store={mockStore}
        params={{ firstName: 'First', lastName: 'Last', dateOfBirth: '09241997' }}
      />
    );

    const phoneNumberInput = getByTestId('phoneNumber');

    fireEvent.changeText(phoneNumberInput, '5104831826');

    expect(toJSON()).toMatchSnapshot();
  });
});
