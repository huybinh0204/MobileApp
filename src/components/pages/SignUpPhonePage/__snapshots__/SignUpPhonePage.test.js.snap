// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPhonePage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="SignUpPhonePage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "width": "50%",
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "center",
                                    "backgroundColor": "#F2F3F4",
                                    "borderColor": "#000000",
                                    "borderRadius": 10,
                                    "height": 9,
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "backgroundColor": "#3D3397",
                                    "borderRadius": 10,
                                    "bottom": 0,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                    "width": "0%",
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 36,
                                    "lineHeight": 50.5,
                                    "marginBottom": 58,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Letâ€™s get started by verifying your phone number
                            </Text>
                            <View
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                flex={1}
                                marginRight={true}
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                      "marginRight": 29,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 87,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "alignItems": "center",
                                          "borderRadius": 6,
                                          "borderWidth": 2,
                                          "flexDirection": "row",
                                          "width": "100%",
                                          "zIndex": 0,
                                        },
                                        Object {
                                          "borderColor": undefined,
                                        },
                                      ]
                                    }
                                  >
                                    <BVLinearGradient
                                      colors={
                                        Array [
                                          309817028,
                                          318208316,
                                        ]
                                      }
                                      endPoint={
                                        Object {
                                          "x": 0.75,
                                          "y": 0.5,
                                        }
                                      }
                                      locations={null}
                                      startPoint={
                                        Object {
                                          "x": 0.25,
                                          "y": 0.5,
                                        }
                                      }
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 6,
                                            "height": "100%",
                                            "position": "absolute",
                                            "width": "100%",
                                            "zIndex": 1,
                                          },
                                          Object {
                                            "opacity": undefined,
                                          },
                                        ]
                                      }
                                    />
                                    <Text
                                      allowFontScaling={false}
                                      onLayout={[Function]}
                                      style={
                                        Array [
                                          Object {
                                            "bottom": 0,
                                            "fontFamily": "Poppins",
                                            "fontSize": 29,
                                            "left": 25.5,
                                            "lineHeight": 40,
                                            "position": "absolute",
                                            "zIndex": 2,
                                          },
                                          Object {
                                            "color": undefined,
                                            "transform": Array [
                                              Object {
                                                "translateY": undefined,
                                              },
                                              Object {
                                                "translateX": undefined,
                                              },
                                              Object {
                                                "scale": undefined,
                                              },
                                            ],
                                          },
                                        ]
                                      }
                                    >
                                      Code
                                    </Text>
                                    <TextInput
                                      autoCapitalize="none"
                                      editable={false}
                                      keyboardType="phone-pad"
                                      onBlur={[Function]}
                                      onChangeText={[Function]}
                                      onFocus={[Function]}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#1F1D25",
                                            "flex": 1,
                                            "fontFamily": "Poppins-SemiBold",
                                            "fontSize": 29,
                                            "lineHeight": 40,
                                            "paddingBottom": 18,
                                            "paddingHorizontal": 25.5,
                                            "paddingTop": 43.5,
                                            "zIndex": 3,
                                          },
                                          undefined,
                                        ]
                                      }
                                      testID="phoneCode"
                                      value="+1"
                                    />
                                  </View>
                                </View>
                              </View>
                              <View
                                flex={4}
                                style={
                                  Array [
                                    Object {
                                      "flex": 4,
                                      "marginRight": 0,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 87,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "alignItems": "center",
                                          "borderRadius": 6,
                                          "borderWidth": 2,
                                          "flexDirection": "row",
                                          "width": "100%",
                                          "zIndex": 0,
                                        },
                                        Object {
                                          "borderColor": undefined,
                                        },
                                      ]
                                    }
                                  >
                                    <BVLinearGradient
                                      colors={
                                        Array [
                                          309817028,
                                          318208316,
                                        ]
                                      }
                                      endPoint={
                                        Object {
                                          "x": 0.75,
                                          "y": 0.5,
                                        }
                                      }
                                      locations={null}
                                      startPoint={
                                        Object {
                                          "x": 0.25,
                                          "y": 0.5,
                                        }
                                      }
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 6,
                                            "height": "100%",
                                            "position": "absolute",
                                            "width": "100%",
                                            "zIndex": 1,
                                          },
                                          Object {
                                            "opacity": undefined,
                                          },
                                        ]
                                      }
                                    />
                                    <Text
                                      allowFontScaling={false}
                                      onLayout={[Function]}
                                      style={
                                        Array [
                                          Object {
                                            "bottom": 0,
                                            "fontFamily": "Poppins",
                                            "fontSize": 29,
                                            "left": 25.5,
                                            "lineHeight": 40,
                                            "position": "absolute",
                                            "zIndex": 2,
                                          },
                                          Object {
                                            "color": undefined,
                                            "transform": Array [
                                              Object {
                                                "translateY": undefined,
                                              },
                                              Object {
                                                "translateX": undefined,
                                              },
                                              Object {
                                                "scale": undefined,
                                              },
                                            ],
                                          },
                                        ]
                                      }
                                    >
                                      Phone
                                    </Text>
                                    <TextInput
                                      autoCapitalize="none"
                                      keyboardType="phone-pad"
                                      maxLength={14}
                                      onBlur={[Function]}
                                      onChangeText={[Function]}
                                      onFocus={[Function]}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#1F1D25",
                                            "flex": 1,
                                            "fontFamily": "Poppins-SemiBold",
                                            "fontSize": 29,
                                            "lineHeight": 40,
                                            "paddingBottom": 18,
                                            "paddingHorizontal": 25.5,
                                            "paddingTop": 43.5,
                                            "zIndex": 3,
                                          },
                                          undefined,
                                        ]
                                      }
                                      testID="phoneNumber"
                                      textContentType="telephoneNumber"
                                      value="(510) 483-1826"
                                    />
                                  </View>
                                </View>
                              </View>
                            </View>
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#D02B35",
                                    "fontFamily": "Poppins",
                                    "fontSize": 25.5,
                                    "lineHeight": 35.5,
                                    "marginTop": 14.5,
                                    "textAlign": "left",
                                  },
                                  undefined,
                                ]
                              }
                            />
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginTop": "auto",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#5F6A77",
                                      "fontFamily": "Poppins",
                                      "fontSize": 20,
                                      "lineHeight": 28,
                                      "marginBottom": 29,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                By submitting your phone number you agree to our [object Object] and to receive future transactional SMS messages & SMS messages about our product & features. Message and data rates may apply. Text STOP to stop receiving messages.
                              </Text>
                            </View>
                            <View
                              accessibilityState={
                                Object {
                                  "disabled": false,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignSelf": "stretch",
                                  "backgroundColor": "#3D3397",
                                  "borderRadius": 10,
                                  "opacity": 1,
                                  "paddingHorizontal": 45.5,
                                  "paddingVertical": 31,
                                }
                              }
                              testID="signUpPhoneSubmitButton"
                            >
                              <Text
                                allowFontScaling={false}
                                disabled={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Continue
                              </Text>
                            </View>
                          </View>
                          <View
                            onLayout={[Function]}
                            paddingHorizontal={null}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "bottom": 0,
                                  "paddingHorizontal": 58,
                                  "position": "absolute",
                                  "width": "100%",
                                },
                                Object {
                                  "opacity": 1,
                                  "transform": Array [
                                    Object {
                                      "translateY": AnimatedValue {
                                        " __value": 0,
                                      },
                                    },
                                  ],
                                },
                              ]
                            }
                            testID="codeErrorToast"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "#FFFFFF",
                                    "borderColor": "#F2F3F4",
                                    "borderRadius": 10,
                                    "borderWidth": 0,
                                    "padding": 29,
                                    "shadowColor": "#362E60",
                                    "shadowOffset": Object {
                                      "height": 7,
                                      "width": 0,
                                    },
                                    "shadowOpacity": 0.1,
                                    "shadowRadius": 10,
                                  },
                                  Array [
                                    Object {
                                      "backgroundColor": "#D02B35",
                                      "borderColor": "#D02B35",
                                      "borderWidth": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ],
                                ]
                              }
                              type="error"
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "flex-start",
                                      "flexDirection": "row",
                                      "justifyContent": "space-between",
                                    },
                                    undefined,
                                  ]
                                }
                                testID="toastHeader"
                              >
                                <SvgMock
                                  height={36}
                                  width={36}
                                />
                                <Text
                                  allowFontScaling={false}
                                  numberOfLines={2}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 21.5,
                                        "lineHeight": 30.5,
                                        "paddingHorizontal": 18,
                                        "paddingTop": 2,
                                      },
                                      undefined,
                                    ]
                                  }
                                  type="error"
                                />
                                <View
                                  accessibilityLabel="Close toast"
                                  accessibilityRole="button"
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                  testID="toastCloseButton"
                                >
                                  <SvgMock
                                    height={20}
                                    style={
                                      Array [
                                        Object {
                                          "height": 65,
                                          "width": 65,
                                        },
                                        undefined,
                                      ]
                                    }
                                    type="error"
                                    width={20}
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
